fe/fe-core/src/main/java/com/starrocks/connector/PartitionUtil.java

```diff
@@ -430,18 +461,32 @@ public static Map<String, Range<PartitionKey>> getRangePartitionMapOfExternalTab
         int partitionColumnIndex = checkAndGetPartitionColumnIndex(partitionColumns, partitionColumn);
         List<PartitionKey> partitionKeys = new ArrayList<>();
         Map<String, PartitionKey> mvPartitionKeyMap = Maps.newHashMap();
-        for (String partitionName : partitionNames) {
-            List<String> partitionNameValues = toPartitionValues(partitionName);
-            PartitionKey partitionKey = createPartitionKey(
-                    ImmutableList.of(partitionNameValues.get(partitionColumnIndex)),
-                    ImmutableList.of(partitionColumns.get(partitionColumnIndex)),
-                    table.getType());
-            partitionKeys.add(partitionKey);
-            String mvPartitionName = generateMVPartitionName(partitionKey);
-            // TODO: check `mvPartitionName` existed.
-            mvPartitionKeyMap.put(mvPartitionName, partitionKey);
+        if (table.isJDBCTable()) {
+            for (String partitionName : partitionNames) {
+                PartitionKey partitionKey = createPartitionKey(
+                        ImmutableList.of(partitionName),
+                        ImmutableList.of(partitionColumn),
+                        table.getType());
+                partitionKeys.add(partitionKey);
+                String mvPartitionName = generateMVPartitionName(partitionKey);
+                // TODO: check `mvPartitionName` existed.
+                mvPartitionKeyMap.put(mvPartitionName, partitionKey);
+            }
+        } else {
+            for (String partitionName : partitionNames) {
```

Comment:
put it in a function

Reply:
Can I make modifications according to this suggestion as a solution to the previous one?

Reply:
done